### PR TITLE
add : wishlist and cart pages with functionality

### PR DESCRIPTION
**Pull request by ApurvaSawant11**
This PR contains Wishlist and Cart pages with basic functionalities.

### Files to Check:
- Wishlist.jsx
- WishlistCard.jsx
- Cart.jsx
- CartSummary.jsx
- CartCard.jsx
- Wishlist and Cart Context Providers
- Wishlist and Cart Reducers

### FIles to ignore:
- CSS files
- files with minor changes (Header, App.js, index.js,...)

_Note: I have used `alert` on the wishlist page. I will use toast in future PRs_ 
#### Live Link : https://dazzle-cerashop-4s9e6epw7-apurvasawant11.vercel.app/

### CODE REVIEW

File: src/components/card/cartCard/CartCard.jsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+  const isInWishlist = isProductInWishlist(wishlist, _id);
+
+  const cartQuantityHandler = async (type) => {
+    await updateCartQuantity(_id, dispatch, token, type);
+  };
+
+  const moveToWishlist = () => {
+    removeFromCart(_id, dispatch, token);
+    !isInWishlist && addToWishlist(wishlistDispatch, productDetails, token);
```
Comment (anik31) [inline src/components/card/cartCard/CartCard.jsx:27]: appreciation - liked that you have divided into small functions

File: src/context/cart-context.jsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+  const value = {
+    cartList: state.cartList,
+    cartDispatch: cartDispatch,
+    addToCart,
+    removeFromCart,
+    updateCartQuantity,
+  };
+  return <CartContext.Provider value={value}>{children}</CartContext.Provider>;
+};
```
Comment (anik31) [inline src/context/cart-context.jsx:89]: appreciation - liked that you're using a separate context for cart & doing the cart related operations inside it

File: src/pages/cart/Cart.jsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      <main>
+        <h4 className="text-center text-uppercase p-1">My Cart</h4>
+        {isCartEmpty ? (
+          <p>
+            Oops! Your cart is empty! 😓. Add product to your cart from{" "}
+            <Link to="/products" className="secondary-text">
+              HERE
+            </Link>
+          </p>
+        ) : (
```
Comment (anik31) [inline src/pages/cart/Cart.jsx:25]: appreciation - good that you're message for empty cart

File: src/reducer/cartReducer.jsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+const cartReducer = (state, action) => {
+  switch (action.type) {
+    case "ADD_TO_CART":
+      return { ...state, cartList: [...action.payload] };
+
+    case "REMOVE_FROM_CART":
+      return { ...state, cartList: [...action.payload] };
+
+    case "UPDATE_CART_QUANTITY":
+      return { ...state, cartList: [...action.payload] };
```
Comment (anik31) [inline src/reducer/cartReducer.jsx]: suggestion - the return value for these reducer are same, so you can replace it with a single reducer, for example "SET_CART"

File: src/reducer/wishlistReducer.jsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+const wishlistReducer = (state, action) => {
+  switch (action.type) {
+    case "ADD_TO_WISHLIST":
+      return { ...state, wishlist: [...action.payload] };
+
+    case "REMOVE_FROM_WISHLIST":
+      return { ...state, wishlist: [...action.payload] };
```
Comment (anik31) [inline src/reducer/wishlistReducer.jsx]: suggestion - same as cart reducer, here also you can use "SET_WISHLIST" beacause return values are same

**Review by anik31 — APPROVED** [review]
Nice work @ApurvaSawant11 💯 Functionality is working fine in the preview & code is clean and easily readable

File: src/reducer/cartReducer.jsx (hunk 4 above)
Comment (ApurvaSawant11) [reply src/reducer/cartReducer.jsx]: Right! I'll make the changes

File: src/reducer/wishlistReducer.jsx (hunk 5 above)
Comment (ApurvaSawant11) [reply src/reducer/wishlistReducer.jsx]: Yes...working on it. Thanks

File: src/components/card/wishlistCard/WishlistCard.jsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+          <span className="discountedPrice">Rs. {discounted}</span>
+          <span className="originalPrice pl-0p5">Rs. {original}</span>
+        </div>
+      </div>
+      <button
+        className="plain-button card-button mt-0p5"
+        onClick={() =>
+          isInCart
+            ? (updateCartQuantity(_id, cartDispatch, token, "INC_QTY"),
+              alert(
```
Comment (jalajcodes) [inline src/components/card/wishlistCard/WishlistCard.jsx:60]: Alert blocks the page and doesn't let the user do anything until they close the alert.
Its fine if you want it like that or you can use maybe a toast in this case.

**Review by jalajcodes — APPROVED** [review]
Just a small comment. Everything else looks good.

File: src/components/card/wishlistCard/WishlistCard.jsx (hunk 6 above)
Comment (ApurvaSawant11) [reply src/components/card/wishlistCard/WishlistCard.jsx:60]: Yes...I have mentioned in the PR that I'll change the alert to toast in later PRs.  Right now I'm focusing on basic functionalities :)

Comment (jalajcodes) [reply src/components/card/wishlistCard/WishlistCard.jsx:60]: Sorry, my bad, didn't notice that.😅

**Pull request merged**